service/service.go

```diff
@@ -137,6 +162,20 @@ func New(config Config) (*Service, error) {
 // Boot starts top level service implementation.
 func (s *Service) Boot() {
 	s.bootOnce.Do(func() {
-		go s.releaseController.Boot(context.Background())
+		ctx := context.Background()
+		backOffFactory := func() backoff.Interface { return backoff.NewMaxRetries(7, 1*time.Second) }
+
+		// Install Release CRD.
+		s.logger.LogCtx(ctx, "level", "debug", "message", "ensuring custom resource definition exists")
+
+		err := s.crdClient.EnsureCreated(ctx, s.crd, backOffFactory())
+		if err != nil {
+			s.logger.LogCtx(ctx, "level", "error", "message", "stop service boot retries due to too many errors", "stack", fmt.Sprintf("%#v", err))
+			os.Exit(1)
+		}
+
+		s.logger.LogCtx(ctx, "level", "debug", "message", "ensured custom resource definition exists")
+
+		go s.releaseController.Boot(ctx)
```

Comment:
I don't remember exact reason for doing that but and I think we shouldn't do that in general. Let's try to skip this `go ` this time and see what happens.

@xh3b4sd maybe you remember what was wrong with calling it synchronously?

Reply:
Well, the controller watches events and `Boot` can always block until it is done and you may don't want that. Everything in `Boot` should be executed concurrently IMO.

Reply:
Oh yeah. Because `Boot` is blocking. We need to get that stuff straight at some point.

Reply:
Does this mean that I should also wrap the crd creation in a separate go routine here ?

Reply:
No I got confused. Forget about it.